### PR TITLE
En 866 entity managers endpoints

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                }
+            }
+        }
+        this.mandatory = attribute.isRequired();
+        this.canBeUsedAsFilterInList = attribute.isSearchable();
+        if (attribute.isSimple()) {
+            return;
+        }
+        /*
+        if (attribute instanceof AbstractListAttribute) {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java]: Can this be deleted?

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            List<AttributeInterface> attributes = ((CompositeAttribute) attribute).getAttributes();
+            for (AttributeInterface compAttribute : attributes) {
+                this.getCompositeAttributes().add(new EntityAttributeDto(compAttribute, roles));
+            }
+        }
+         */
+    }
+
+    /*
+"attributes": [{
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java]: Can this be deleted?

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    public boolean isCanBeUsedAsFilterInList() {
+        return canBeUsedAsFilterInList;
+    }
+
+    public void setCanBeUsedAsFilterInList(boolean canBeUsedAsFilterInList) {
+        this.canBeUsedAsFilterInList = canBeUsedAsFilterInList;
+    }
+    /*
+    public EntityAttributeDto getNestedAttribute() {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java]: Delete?

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityManagerDto.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        return entityTypes;
+    }
+
+    public void setEntityTypes(List<EntityTypeDto> entityTypes) {
+        this.entityTypes = entityTypes;
+    }
+
+}
+
+/*
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityManagerDto.java]: Delete?

File: engine/src/main/java/org/entando/entando/web/entity/EntityManagerController.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public void setEntityManagerService(IEntityManagerService entityManagerService) {
+        this.entityManagerService = entityManagerService;
+    }
+
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getEntityManagers(RestListRequest requestList) {
+        PagedMetadata<String> result = this.getEntityManagerService().getEntityManagers(requestList);
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/entity/EntityManagerController.java]: We should put some debug logging in these controllers with the payload.

File: engine/src/main/java/org/entando/entando/web/entity/EntityManagerController.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        return new ResponseEntity<>(new RestResponse(result.getBody(), null, result), HttpStatus.OK);
+    }
+
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RequestMapping(value = "/{entityManagerCode}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getEntityManager(@PathVariable String entityManagerCode) {
+        EntityManagerDto dto = this.getEntityManagerService().getEntityManager(entityManagerCode);
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/entity/EntityManagerController.java]: We should put some debug logging in these controllers with the payload.

Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/web/entity/EntityManagerController.java]: Added log (debug) for request and response

File: engine/src/main/java/org/entando/entando/web/entity/EntityManagerController.java (hunk 5 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/web/entity/EntityManagerController.java]: Added log (debug) for response

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityManagerDto.java (hunk 4 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityManagerDto.java]: Done

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java (hunk 3 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java]: Done

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java (hunk 2 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java]: Done

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java (hunk 1 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java]: Done

**Comment by entando-service** [comment]
All is well

**Pull request merged**